### PR TITLE
Reintroduce `startNode` and `endNode` on `ChildPart`.

### LINKED ISSUE #1641: [lit-next] Reintroduce `startNode` and `endNode` to `ChildPart`
`NodePart` from Lit 1 had `startNode` and `endNode` properties that allowed reaching into the DOM rendered by the part. `ChildPart` should also have these properties.

### PR DESCRIPTION
**Pull request by bicknellr**
This PR exposes the start and end marker nodes of `ChildPart` through `startNode` and `endNode` getters, like `NodePart` in Lit 1.

Fixes #1641.

### CODE REVIEW

File: packages/lit-html/src/lit-html.ts [hunk 1]
```diff
@@ -909,10 +909,46 @@ class ChildPartImpl {
...
   get parentNode(): Node {
     return wrap(this._$startNode).parentNode!;
   }
 
+  /**
+   * The part's leading marker node, if any. See `.parentNode` for more
+   * information.
+   */
+  get startNode(): Node | undefined {
+    return this._$startNode === this._$endNode ? undefined : this._$startNode;
```
Comment (kevinpschaaf) [inline packages/lit-html/src/lit-html.ts:939]: Just return startNode/endNode for now, but this protects for doing what you wrote in the future.

**Review by kevinpschaaf — CHANGES_REQUESTED** [review]

File: packages/lit-html/src/lit-html.ts (hunk 1 above)
Comment (bicknellr) [reply packages/lit-html/src/lit-html.ts]: Ok, updated.

File: packages/lit-html/src/lit-html.ts [hunk 2]
```diff
@@ -911,10 +911,44 @@ class ChildPartImpl {
...
+  get startNode(): Node | undefined {
+    return this._$startNode;
+  }
+
+  /**
+   * The part's trailing marker node, if any. See `.parentNode` for more
+   * information.
+   */
+  get endNode(): Node | undefined {
+    return this._$endNode ?? undefined;
```
Comment (bicknellr) [inline packages/lit-html/src/lit-html.ts:949]: Would you prefer I change the `| undefined` in the return types to `| null` also? I guess it's DOM convention and would let us avoid `?? undefined`.

Comment (bicknellr) [reply packages/lit-html/src/lit-html.ts:949]: I guess I typically prefer using `undefined` as the 'true' null value thing, but this seems like a case where we could go the other way.

Comment (justinfagnani) [reply packages/lit-html/src/lit-html.ts:949]: It's an interesting call. In JS `undefined` is the missing value, and in the DOM `null` is. 🤬

One reason why `null` flows in here so naturally is that `null` is returned from `node.nextSibling` and a valid value for `insertBefore()`'s ref node. I'd say in this case that we don't try to force the natural type of this into our default style and type the getters as `Node | null`.

Comment (bicknellr) [reply packages/lit-html/src/lit-html.ts]: Ok, updated, PTAL

File: packages/lit-html/src/lit-html.ts [hunk 3]
```diff
@@ -911,10 +911,44 @@ class ChildPartImpl {
...
+   * `.parentNode`, possibly bordered by 'marker nodes' (`.startNode` and
+   * `.endNode`).
+   *
+   * - If both `.startNode` and `.endNode` are defined, then the part's content
+   * consists of all siblings between `.startNode` and `.endNode`, exclusively.
+   *
+   * - If `.startNode` is defined but `.endNode` is undefined, then the part's
+   * content consists of all siblings following `.startNode`, up to and
+   * including the last child of `.parentNode`. (If `.endNode` is defined, then
+   * `.startNode` will always be defined.)
```
Comment (justinfagnani) [inline packages/lit-html/src/lit-html.ts:927]:
```suggestion
   * - If `.startNode` is defined but `.endNode` is null, then the part's
   * content consists of all siblings following `.startNode`, up to and
   * including the last child of `.parentNode`. If `.endNode` is non-null, then
   * `.startNode` will always be non-null.
```

**Review by justinfagnani — APPROVED** [review]

**Pull request merged**